tools/src/commands/RunStories.ts

```diff
@@ -0,0 +1,241 @@
+import spawnAsync from '@expo/spawn-async';
+import { IosPlist } from '@expo/xdl';
+import fs from 'fs';
+import inquirer from 'inquirer';
+import path from 'path';
+import { podInstallAsync } from '../CocoaPods';
+
+import { runExpoCliAsync } from '../ExpoCLI';
+import Logger from '../Logger';
+
+type Action = {
+  platform: 'android' | 'ios' | 'web';
+  rebuild: boolean;
+  clearCache: boolean;
+};
+
+async function action(packageName: string, options: Action) {
```

Comment:
General remark to this function: it's too long! Far too long!
Try to refactor it into multiple, single-purpose ones. I imagine sth like this (you can probably name the steps far better than I).
```ts
async function action(...) {
   const packageName = await getPackageName();
   await ensureStoryLoaderExists();
   await possiblyRebuildStoryLoader();
   await Promise.all([configureIosProject(), configureAndroidProject(), configureWebProject()]; // these could rather be done simultaneously?
   etc...
}
```
Ideally I will get quite short but very descriptive top-level function and then I can dig deeper into the more atomic ones, but I still expect simplicity and rather single-purpose shortness.